### PR TITLE
tags search only in navbar

### PR DESCRIPTION
**Pull request by jywarren**
Re: https://github.com/publiclab/plots2/issues/5719

### CODE REVIEW

**Comment by jywarren** [comment]
```
 FAIL["test_searching_using_navbar_autocomplete", #<Minitest::Reporters::Suite:0x00007f2b88a8e5e8 @name="SearchTest">, 93.96072765900004]
 test_searching_using_navbar_autocomplete#SearchTest (93.96s)
        expected to find visible css ".typeahead li" with text "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON" but there were no matches. Also found "View all", which matched the selector but not all filters. 
        test/system/search_test.rb:19:in `block in <class:SearchTest>'
```

**Comment by jywarren** [comment]
Hmm:

```

 FAIL["test_viewing_the_dashboard", #<Minitest::Reporters::Suite:0x00005609ff937650 @name="DashboardTest">, 64.75549231800005]
 test_viewing_the_dashboard#DashboardTest (64.76s)
        expected to find visible css ".row.header > h1" with text "Dashboard" but there were no matches. Also found "Community research", which matched the selector but not all filters. 
        test/system/dashboard_test.rb:19:in `block in <class:DashboardTest>'
[Screenshot]: tmp/screenshots/failures_test_searching_using_navbar_autocomplete.png
 FAIL["test_searching_using_navbar_autocomplete", #<Minitest::Reporters::Suite:0x0000560a035b0140 @name="SearchTest">, 153.71914509099997]
 test_searching_using_navbar_autocomplete#SearchTest (153.72s)
        expected to find visible css ".typeahead li" with text "kites" but there were no matches. Also found "View all", which matched the selector but not all filters. 
```

File: test/system/search_test.rb [hunk 1]
```diff
@@ -14,11 +14,11 @@ class SearchTest < ApplicationSystemTestCase
   test 'searching using navbar autocomplete' do
     visit '/'
 
-    fill_in("searchform_input", with: "Canon")
+    fill_in("searchform_input", with: "Kite")
 
-    assert_selector ".typeahead li", text: "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON", wait: 10
+    assert_selector ".typeahead li", text: "kites", wait: 10
```
Comment (jywarren) [inline test/system/search_test.rb]: `expected to find visible css ".typeahead li" with text "kites" but there were no matches. Also found "View all", which matched the selector but not all filters. `

File: test/system/screenshots_test.rb [hunk 2]
```diff
@@ -5,8 +5,11 @@ class ScreenshotsTest < ApplicationSystemTestCase
 
   test 'front page with navbar search autocomplete' do
     visit '/'
+
     fill_in("searchform_input", with: "Canon")
-    assert_selector ".typeahead li", text: "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON"
+    
+    assert_selector ".typeahead li", text: "Canon A1200 IR conversion at PLOTS Barnraising at LUMCON", wait: 10
```
Comment (jywarren) [inline test/system/screenshots_test.rb]: We'll have to modify this for tags search only, as well.

**Comment by jywarren** [comment]
Are regular tests really still running? https://travis-ci.org/publiclab/plots2/jobs/544892242#L3984 cc @alaxalves

**Comment by jywarren** [comment]
Let's just wait for #5885 and then rebase this over it...

**Comment by jywarren** [comment]
FAIL["test_search_tags_functionality", #<Minitest::Reporters::Suite:0x0000564b6c646398 @name="SearchApiTest">, 66.52781274400002]
 test_search_tags_functionality#SearchApiTest (66.53s)
        Expected: "/tag/test"
          Actual: "/test"
        test/functional/api/search_api_test.rb:126:in `block in <class:SearchApiTest>'

**Comment by jywarren** [comment]
FAIL["test_searching_using_navbar_autocomplete", #<Minitest::Reporters::Suite:0x000055b55622c390 @name="SearchTest">, 22.565934336999987]
 test_searching_using_navbar_autocomplete#SearchTest (22.57s)
        expected to find visible css ".typeahead li" with text "kites" but there were no matches. Also found "View all", which matched the selector but not all filters. 
        test/system/search_test.rb:19:in `block in <class:SearchTest>'

**Comment by jywarren** [comment]
😭 i don't know how this would be related at all to these changes!!!

```
 FAIL["test_graph", #<Minitest::Reporters::Suite:0x0000559f56c4c5e8 @name="TagSelectionTest">, 0.3037873540000078]
 test_graph#TagSelectionTest (0.30s)
        Expected: 16
          Actual: 20
        test/unit/tag_selection_test.rb:12:in `block in <class:TagSelectionTest>'

```

**Comment by jywarren** [comment]
@publiclab/reviewers, I'm pretty stuck here! Can anyone shed light on this test issue? How weird!

**Comment by jywarren** [comment]
https://github.com/publiclab/plots2/blob/a0f77722688be921e1207d40091beb528c667d3e/test/unit/tag_selection_test.rb#L12

confusing

**Comment by CoderJolly** [comment]
> https://github.com/publiclab/plots2/blob/a0f77722688be921e1207d40091beb528c667d3e/test/unit/tag_selection_test.rb#L12
> 
> confusing

I see that the PR is merged, so you found the solution ? Kindly share it 😄

**Comment by jywarren** [comment]
Thanks for the reminder - i should have!

OK, I actually didn't do anything but update with the master branch. But i have a couple theories:

1. I wonder if the containers are running but are all using the same database container? And now that all are running at the same time, we have some odd behavior resulting? but this seems unlikely.
2. This PR could certainly have caused more subscriptions: https://github.com/publiclab/plots2/pull/5928 but i don't see how it could have related to this one. 
3. could there be some time based non-deterministic behavior because we're using Time.now twice very close to one another, and perhaps we're sometimes getting very recently made `tag_selections` (within 1 second) as all the tests run real fast, but other times they are not being selected because they don't occur in the same second? We've used Timecop in some tests to address this kind of failure: https://github.com/publiclab/plots2/blob/5c0d96274bd0be15feba69d8f30093082402ace5/test/functional/features_controller_test.rb#L12

Have you seen the same error anywhere else?

**Comment by CoderJolly** [comment]
> Thanks for the reminder - i should have!
> 
> OK, I actually didn't do anything but update with the master branch. But i have a couple theories:
> 
> 1. I wonder if the containers are running but are all using the same database container? And now that all are running at the same time, we have some odd behavior resulting? but this seems unlikely.
> 2. This PR could certainly have caused more subscriptions: #5928 but i don't see how it could have related to this one.
> 3. could there be some time based non-deterministic behavior because we're using Time.now twice very close to one another, and perhaps we're sometimes getting very recently made `tag_selections` (within 1 second) as all the tests run real fast, but other times they are not being selected because they don't occur in the same second? We've used Timecop in some tests to address this kind of failure: https://github.com/publiclab/plots2/blob/5c0d96274bd0be15feba69d8f30093082402ace5/test/functional/features_controller_test.rb#L12
> 
> Have you seen the same error anywhere else?

No @jywarren , I haven't seen this anywhere else...bt I wanted to know more about from ur perspective. Thanks for the insight

**Pull request merged**